### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by lbayle** [mantisbt-plugins/Taskodrome#246]
Hi,
Firefox complains avout a page slowing down the browser. 
It takes a long time to display the "Assignment board"  (+200 issues to display)

Very promissing tool, we already love it !

**Comment by thanhhungtn1** [comment]
+1 for this issue, we have 280 tickets under the filter of My View page and it took forever to load. 
Please fix this issue

**Comment by thanhhungtn1** [comment]
Further investigation is I only filtered out the status and the other filtered set to default. May some issue with the selecting datas queries?

**Comment by nenadalm** [comment]
https://github.com/mantisbt-plugins/Taskodrome/blob/58a15c5f369bfdaa9dbc6696660cc4a1b6b9cdd0/Taskodrome/pages/main.php#L34 is causing for me to that over 32 000 records per page is configured. Main document loads in 2.1 minutes and there are errors in js console, missing buttons on page and I see no tickets.

When I change `$t_per_page` to `null` it causes configuration of `50` records per page, main document loads in 7s but it still takes way longer to show some tickets. Records per page probably shouldn't be unlimited?

**Comment by nenadalm** [comment]
Related to js performance I found that when commenting out this `while`: https://github.com/mantisbt-plugins/Taskodrome/blob/23af0e093f940355fedffcf3401af54eb56b4710/Taskodrome/files/scripts/utils.js#L106 rendering time drops by ~1min.

**Comment by nenadalm** [comment]
In case you're interested I started working on performance improvements: https://github.com/imatic/Taskodrome/commits/imatic_mantis2 for now js renders page with 50 records about 1minute faster (many long strings which this plugin was shortening quite a while).

**Comment by AuthenticEshkinKot** [comment]
Hello, @nenadalm !
Thank you for help! I'll check out your commits.

**Comment by nenadalm** [comment]
Do I understand it correctly that this plugin just renders all issues in mantis in selected project? So that if I have over 32000 issues, they will all be rendered at once (status doesn't matter) and there is no way to configure it?

**Comment by AuthenticEshkinKot** [comment]
Yes, this plugin gets all available issues of current project and tries to show them. Admin can hide some users or statuses with their issues using plugin configuration.

**Comment by nenadalm** [comment]
Plugin configuration won't affect number of rows from database though: https://github.com/mantisbt-plugins/Taskodrome/blob/58a15c5f369bfdaa9dbc6696660cc4a1b6b9cdd0/Taskodrome/pages/main.php#L38 right? So no matter what I configure, plugin will always fetch all 32 000 issues?

**Comment by AuthenticEshkinKot** [comment]
Yep, that's right

**Comment by nenadalm** [comment]
Ok, thanks. It turned out that this plugin is not something we would like to use. Feel free to merge in the performance improvements in case of tasks with long texts. I won't look more into this.

**Comment by AuthenticEshkinKot** [comment]
@nenadalm 
Could you make a pull request with your improvements, please? It's your work, so it would be more fair if your account would be attached to it.

**Comment by nenadalm** [comment]
I don't think I can. I am getting redirections when trying to create PR: https://github.com/mantisbt-plugins/Taskodrome/compare/mantisbt-plugins:mantisbt_2.0.0...imatic:imatic_mantis2?expand=1 back to the create PR page. I don't really mind which account is attached to the changes :)

**Comment by AuthenticEshkinKot** [comment]
OK :)
